internal/gitcontext/branchfiles.go

```
@@ -0,0 +1,157 @@
+type RunGitFunc func(root string, env []string, args ...string) ([]byte, error)
+
+// Options configures Collect. A nil RunGit uses the real git binary.
+type Options struct {
+	Base       string
+	Excludes   []string
+	Exceptions []string
+	RunGit     RunGitFunc
+}
+
+func realGit(root string, env []string, args ...string) ([]byte, error) {
+	cmd := exec.Command("git", args...)
```

golangci-lint (2.12.2)

[medium] G204: Subprocess launched with variable

(gosec)